src/core/Header/Header.tsx

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import SvgHelpCircularHollow from '@itwin/itwinui-icons-react/cjs/icons/HelpCircularHollow';
+import SvgMoreVertical from '@itwin/itwinui-icons-react/cjs/icons/MoreVertical';
+import SvgNotification from '@itwin/itwinui-icons-react/cjs/icons/Notification';
+
+import cx from 'classnames';
+import React from 'react';
+import { IconButton } from '../Buttons';
+
+import { useTheme } from '../utils/hooks/useTheme';
+import { DropdownMenu } from '../DropdownMenu';
+
+export type HeaderTranslations = {
+  notifications: string;
+  help: string;
+  myAccount: string;
+  moreOptions: string;
+};
+
+export type HeaderProps = {
+  /**
+   * Icon to the application.
+   */
+  appIcon: React.ReactNode;
+  /**
+   * Name of the application.
+   */
+  appName?: React.ReactNode;
+  /**
+   * Handler on application click
+   */
+  onApplication?: () => void;
+  /**
+   * `HeaderButton` array, chevron will be added between each.
+   */
+  breadcrumbsButtons?: React.ReactNode[];
+  /**
+   * Content to display in the center section of the header, should handle iui-slim class if used.
+   */
+  centerContent?: React.ReactNode;
+  /**
+   * Expects `UserIcon`, will be given a drop down if `userMenuItems` is also provided;
+   */
+  userIcon?: React.ReactNode;
+  /**
+   * Handler on notification click, will show notification bell if provided.
+   */
+  onNotification?: () => void;
```

Comment:
This is only a handler instead of a `menuItems` list, because typically the bell button would open a side panel with the notifications (for which we might want to create a component eventually :) )

Reply:
Not sure we need a top level prop for that

Reply:
Well, we dont control the notification panel in the Header, how would the application know that the bell was clicked and that it should show that panel if we dont provide a click handler ?

Reply:
Oh you mean the actual bell icon click...

To me, "on notification click" implies that an actual notification item (from the panel) is clicked. Since that panel is not part of this component, I would actually remove this prop completely.

In #32 I suggested grouping the "right" part of header where the user can provide any arbitrary items. So you could do that, and leave it upto the user to provide both the bell icon and click handler

Reply:
Ok, I misunderstood @veekeys point in #32 , I can change that to only control User icon and More, but that means we'll have no way of smartly collapsing stuff when space is not available, at this point, not sure it's even worth trying to handle User and  More either, as the only thing that this is adding is the `iui-user-icon`, to override `iui-user-icon-medium` while in slim mode, which, if we follow the class names split, should probably be `iui-user-icon iui-medium` anyway, in which case we can decide to do that override without having to "add" the class at all...

Reply:
Actually, we dont even have to add that class, as we could simply override `iui-user-icon-medium` if it's in the header...

Reply:
regarding the `iui-user-icon` that have already been changed in the CSS PR to use the new names, so technically, we wouldn't need to control any of the content of `iui-center` and `iui-right`. With some discussions, I'm starting to think that maybe we could reduce the complexity of this comopnent to `<Header leftContent centerContent rightContent />`, these contents being
```
<Header
  leftContent={
  <>
    <ApplicationButton />
    <Breadcrumb>
      <HeaderButton "project" />
      <HeaderButton "model" />
    </Breadcrumb>
  </>}
  centerContent={<Whatever />}
  rightContent={
  <>
    <IconButton><SvgNotification/></IconButton>
    <button className="iui-button iui-borderless" style={{pointerEvents: 'none'}}>
      <UserIcon />
    </button>
  </>}
/>
```

This way, they have full control over what is in their menu, we dont force anyone, and in a second time, we want to have a component that do control all this, along with collapsing to a full "..." menu if there is not enough space, we can do that, and they will be able to use it without breaking anyone.

Reply:
At that point, it doesn't even make sense to have props, and everything should be handled using `children` by providing granular components. Kinda like what material ui does: https://material-ui.com/components/cards/#api

But their reasoning is that their library needs to be very flexible. Our library is the opposite: enforce standards first, provide flexibility second.

We could continue this discussion in #32 or offline (over call) but I really think there is some middle ground where we don't provide 3 huge props and we don't provide 15 smaller props.

Reply:
Changed the API in may ways, will resolve this, we can create a different discussion if needed.